### PR TITLE
[WIP] Release 6.2

### PR DESCRIPTION
**Pull request by andreasohlund**
Target release date: Match 28

### Plan of attack (TBD)

- [x] Create release branch
- [x] Put RC1 on MyGet
- [x] Internal announcement
- [ ] Smoke test doco
- [ ] Finalize public annoncement: https://github.com/Particular/Announcements/pull/124
- [x] [Update msmq wire compat tests](https://github.com/Particular/NServiceBus.Msmq.WireCompatTests/pull/5) - @andreasohlund
- [ ] Once released verify that serializer tests auto trigger
- [ ] Upgrade guide for the acceptance tests changes and ping downstream maintainers @timbussmann: https://github.com/Particular/NServiceBus/pull/4578#issuecomment-289439439

### CODE REVIEW

**Comment by andreasohlund** [comment]
Wirecompat tests for MSMQ runs green now

**Comment by timbussmann** [comment]
Acceptance test upgrade guide:

* update nservicebus.acceptancetests package.
* Add a new `public partial` class named `TestSuiteConstraints` in the `NServiceBus.AcceptanceTests` namespace to the acceptance tests project.
* Implement the missing members on the added `TestSuiteConstraints` class:
  * Specify the configurations this test suite supports (replaces `IConfigureSupportedScenariosForTestExecution.UnsupportedScenarioDescriptorTypes`)
  * Assign the transport configuration class (e.g. `ConfigureEndpointMsmqTransport`) to the `TransportConfiguration` property.
  * Assign the persistence configuration class (e.g. `ConfigureEndpointInMemoryPersistence`) to the `PersistenceConfiguration` property.
* Delete the implementation of `IConfigureSupportedScenariosForTestExecution`
*. Replace usages of `AddMapping` with `endpointConfiguration.ConfigureTransport().Routing().RouteToEndpoint(typeof(TMessage), typeof(TEndpoint));`
* Replace usages of `AuditTo<T>()` with `endpointConfiguration.AuditProcessedMessagesTo<T>()`.

**Comment by bording** [comment]
FYI, from what I can tell, the reason we have merge conflicts here is because of our decision not to merge hotfix branches on master back into develop.

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -1625,7 +1785,7 @@ namespace NServiceBus.Extensibility
 {
     public class ContextBag : NServiceBus.Extensibility.ReadOnlyContextBag
     {
-        public ContextBag(NServiceBus.Extensibility.ContextBag parentBag = null) { }
+        public ContextBag(NServiceBus.Extensibility.ContextBag parent = null) { }
```
Comment (timbussmann) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: potential breaking change

**Comment by bording** [comment]
Release branch fully merged and bits have been pushed to MyGet.

**Comment by bording** [comment]
Ok, I've gone ahead and pushed up to nuget.org as well.

We do still need to go over the generated release notes and see if there's anything we want to remove.

After that, we should be able to finish up the announcement and post it.

**Pull request merged**